### PR TITLE
Feature: Request list page

### PR DESCRIPTION
**Pull request by Duc-Lam-jh**

### CODE REVIEW

File: src/components/DeclareForm/DeclareForm.js [hunk 1]
```diff
@@ -29,6 +28,7 @@ class DeclareForm extends Component {
   }
 
   handleAddImage = (images) => {
+    console.log('length: ', images.length);
```
Comment (anh-dinh-jh) [inline src/components/DeclareForm/DeclareForm.js]: All console.log should be eliminated as it's not a good practice and make the code ugly

File: src/components/DeviceInfo/DesktopInfo.js [hunk 2]
```diff
@@ -27,4 +28,17 @@ const DesktopInfo = (props) => {
...
+  detail: {
+    brand: 'Chưa có thông tin',
+    configuration: 'Chưa có thông tin',
+    size: 'Chưa có thông tin',
+    images: []
+  },
+}
+
+DesktopInfo.propTypes = {
+  detail: PropTypes.object.isRequired
```
Comment (anh-dinh-jh) [inline src/components/DeviceInfo/DesktopInfo.js]: What if I pass the wrong type of `brand`, `configuration`, `size`, `images`?

File: src/components/DeviceInfo/LaptopInfo.js [hunk 3]
```diff
@@ -27,4 +28,17 @@ const LaptopInfo = (props) => {
...
+  detail: {
+    brand: 'Chưa có thông tin',
+    configuration: 'Chưa có thông tin',
+    seriesNumber: 'Chưa có thông tin',
+    images: []
+  },
+}
+
+LaptopInfo.propTypes = {
+  detail: PropTypes.object.isRequired
```
Comment (anh-dinh-jh) [inline src/components/DeviceInfo/LaptopInfo.js]: Same as the above comment

File: src/components/DeviceInfo/MouseInfo.js [hunk 4]
```diff
@@ -21,4 +22,15 @@ const MouseInfo = (props) => {
...
 
+MouseInfo.defaultProps = {
+  detail: {
+    brand: 'Chưa có thông tin',
+    images: []
+  },
+}
+
+MouseInfo.propTypes = {
+  detail: PropTypes.object.isRequired
```
Comment (anh-dinh-jh) [inline src/components/DeviceInfo/MouseInfo.js]: Same as the above comment

File: src/components/DeviceInfo/UserInfo.js [hunk 5]
```diff
@@ -16,4 +17,16 @@ const UserInfo = (props) => {
...
 
+UserInfo.defaultProps = {
+  detail: {
+    name: 'Chưa có thông tin',
+    email: 'Chưa có thông tin',
+    team: 'Chưa có thông tin'
+  }
+}
+
+UserInfo.propTypes = {
```
Comment (anh-dinh-jh) [inline src/components/DeviceInfo/UserInfo.js:28]: Same as the above comment

File: src/components/ImageInput/ImageInput.js [hunk 6]
```diff
@@ -15,6 +15,7 @@ class ImageInput extends Component {
 
   componentDidUpdate(prevProps, prevState) {
     if(prevState.images.length !== this.state.images.length){
+      // console.log(this.state.images.length);
```
Comment (anh-dinh-jh) [inline src/components/ImageInput/ImageInput.js]: Same as the above comment

File: src/components/ItemEdit/ItemEdit.js [hunk 7]
```diff
@@ -25,4 +27,20 @@ const ItemEdit = (props) => {
...
+    configuration: 'Chưa có thông tin',
+    size: 'Chưa có thông tin',
+    description: 'Chưa có thông tin',
+    seriesNumber: 'Chưa có thông tin',
+  },
+  images: []
+}
+
+ItemEdit.propTypes = {
+  detail: PropTypes.object.isRequired,
```
Comment (anh-dinh-jh) [inline src/components/ItemEdit/ItemEdit.js]: Same as above comment

File: src/components/RequestList/RequestList.js [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+            <div className='title'>Requester: {request.user.name}</div>
+            <div className='subtitle'>{request.reason}</div>
+          </div>
+          <div className='controls'>
+            <button className='accept' onClick={() => handleChangeRequestStatus(request, 'accepted')}>
+              Accept
+            </button>
+            <button className='decline' onClick={() => handleChangeRequestStatus(request, 'declined')}>
+              Decline
+            </button>
```
Comment (anh-dinh-jh) [inline src/components/RequestList/RequestList.js]: Shouldn't have hard-code value here: `accepted`, `declined`

File: src/pages/RequestListPage/RequestListPage.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    const getPendingRequests = async () => {
+      const uri = process.env.REACT_APP_BASE_API_URL + '/requests?status=pending';
+      const response = await fetch(uri);
+      const requestData = await response.json();
+
+      for(let i = 0; i < requestData.length; i++) {
+        requestData[i].user = await getUser(requestData[i].userId);
+      }
```
Comment (anh-dinh-jh) [inline src/pages/RequestListPage/RequestListPage.js:29]: Does this one make the web slower if we have to wait for every request to be processed step by step?

File: src/components/DeclareForm/DeclareForm.js (hunk 1 above)
Comment (Duc-Lam-jh) [reply src/components/DeclareForm/DeclareForm.js]: Sorry I forgot to delete this

File: src/components/DeviceInfo/DesktopInfo.js (hunk 2 above)
Comment (Duc-Lam-jh) [reply src/components/DeviceInfo/DesktopInfo.js]: Noted, I will fix this by adding PropTypes.shape

File: src/pages/RequestListPage/RequestListPage.js (hunk 9 above)
Comment (Duc-Lam-jh) [reply src/pages/RequestListPage/RequestListPage.js:29]: It does make the user wait for the action to finish, right now I am displaying a "loading..." message when the data is being fetched. To my knowledge this is the general practice when fetching data for a component. If there is any better way to improve the performance please let me know  😄

**Pull request merged**